src/QuickCheckVEngine/RVFI_DII.hs

```diff
@@ -37,62 +38,116 @@
 -- SUCH DAMAGE.
 --
 
-{-|
-    Module      : QuickCheckVEngine.RVFI_DII
-    Description : The RVFI-DII interface
-
-    This module re-exports the 'QuickCheckVEngine.RVFI_DII.RVFI' and
-    'QuickCheckVEngine.RVFI_DII.DII' modules, and provides functions to send and
-    receive 'DII_Packet's and 'RVFI_Packet's over a 'Socket'.
--}
-
+-- |
+--    Module      : QuickCheckVEngine.RVFI_DII
+--    Description : The RVFI-DII interface
+--
+--    This module re-exports the 'QuickCheckVEngine.RVFI_DII.RVFI' and
+--    'QuickCheckVEngine.RVFI_DII.DII' modules, and provides functions to send and
+--    receive 'DII_Packet's and 'RVFI_Packet's over a 'Socket'.
 module QuickCheckVEngine.RVFI_DII (
   module QuickCheckVEngine.RVFI_DII.RVFI
 , module QuickCheckVEngine.RVFI_DII.DII
+, RvfiDiiConnection(..)
 , sendDIIPacket
 , sendDIITrace
-, recvRVFIPacket
 , recvRVFITrace
+, rvfiNegotiateVersion
 ) where
 
-import QuickCheckVEngine.RVFI_DII.RVFI
-import QuickCheckVEngine.RVFI_DII.DII
-
-import Data.Int
-import Data.Binary
 import Control.Monad
+import Data.Binary
+import Data.Binary.Get
+import qualified Data.ByteString.Lazy as BS
+import Data.Int
 import Network.Socket
 import Network.Socket.ByteString.Lazy
-import qualified Data.ByteString.Lazy as BS
+import QuickCheckVEngine.RVFI_DII.DII
+import QuickCheckVEngine.RVFI_DII.RVFI
+
+data RvfiDiiConnection = RvfiDiiConnection Socket Int String Int
```

Comment:
Hm, thinking about this more, verbosity shouldn't really be part of the connection

Reply:
You could image adding per-implementation verbosity in the future (e.g. I want debug output for impl A but not B if you're debugging bad packets being sent).

Somehow we need to forward verbosity to the parsing code, and including it in the connection seemed like the most straghtforward way.

Reply:
Let me see if I can easily remove this.

Reply:
Removed in the latest commit.